src/Value/Value.php

```diff
@@ -66,24 +66,31 @@ public static function parseValue(ParserState $oParserState, array $aListDelimit
             $oParserState->consumeWhiteSpace();
         }
         // Convert the list to list objects
+        $iStackLength = count($aStack);
         foreach ($aListDelimiters as $sDelimiter) {
```

Comment:
I think `$iStackLength` needs to be set at the start of the outer loop, not before it.  The stack is updated at the end of the outer loop with `$aStack = $aNewStack`, and may now be shorter for the next iteration if some elements could be joined for the specific delimiter.

Reply:
Yep, you're right. Fixed.